CT-1495 Limit scope for SAR cases

stephenrichards opened the pull request:
The scope for SAR cases is limited as follows:

 - Managers can see all SAR cases
 - responders can only see SAR cases assigned to their team
 - approvers can not see any SAR cases (this will change when we
   introduce trigger sars: Disclosure will be able to see trigger SARS
   but not press or private)

This is done by having the #resole method for Case::BasePolicy::Scope call

aliuk2012 commented on spec/views/cases/incoming_cases_html_slim_spec.rb:
```diff
@@ -18,7 +18,7 @@
     case1
     case2
-    assign(:cases, PaginatingDecorator.new(Case.all.page.order(:received_date)))
+    assign(:cases, PaginatingDecorator.new(Case.all.page.order(:id)))
```
As the received date were exactly the same it really depending on how Rails sorted the list which meant it was not predictable. So sorting the cases by ID meant that the order would be consistent and the content could be tested

stephenrichards committed:
CT-1495 Limit scope for SAR cases

The scope for SAR cases is limited as follows:

 - Managers can see all SAR cases
 - responders can only see SAR cases assigned to their team
 - approvers can not see any SAR cases (this will change when we
   introduce trigger sars: Disclosure will be able to see trigger SARS
   but not press or private)

This is done by having the #resole method for Case::BasePolicy::Scope call

stephenrichards committed:
CT-1459 Tidy up show? policies for FOI standard and SAR cases

This change ensures that responders cannot view SAR cases (this
will be changed when we introduce trigger SARS to allow disclosure
specialist to view trigger SAR cases,  but not press/ private.

It also replaces the #show? method on the Case::BasePolicy to not allow
anyone to virew.  This is handled by the policies for Case::FOI::Standard
and Case::SAR

stephenrichards committed:
CT-1495 Extend responder scope to include all FOI cases
